src/mimefactory.rs

```diff
@@ -1174,18 +1212,58 @@ impl MimeFactory {
                 Loaded::Mdn { .. } => true,
             };
 
-            // Encrypt to self unconditionally,
-            // even for a single-device setup.
-            let mut encryption_keyring = vec![encrypt_helper.public_key.clone()];
-            encryption_keyring.extend(encryption_keys.iter().map(|(_addr, key)| (*key).clone()));
+            let shared_secret: Option<String> = match &self.loaded {
+                Loaded::Message { chat, msg }
+                    if should_encrypt_with_broadcast_secret(msg, chat) =>
+                {
+                    let secret = load_broadcast_shared_secret(context, chat.id).await?;
+                    if secret.is_none() {
+                        // If there is no shared secret yet
+                        // because this is an old broadcast channel,
+                        // created before we had symmetric encryption,
+                        // we show an error message.
+                        let text = r#"The up to now "experimental channels feature" is about to become an officially supported one. By that, privacy will be improved, it will become faster, and less traffic will be consumed.
+
+As we do not guarantee feature-stability for such experiments, this means, that you will need to create the channel again. 
+
+Here is what to do:
+ • Create a new channel
+ • Tap on the channel name
+ • Tap on "QR Invite Code"
+ • Have all recipients scan the QR code, or send them the link
+
+If you have any questions, please send an email to delta@merlinux.eu or ask at https://support.delta.chat/."#;
+                        chat::add_info_msg(context, chat.id, text, time()).await?;
+                        bail!(text);
+                    }
+                    secret
+                }
+                _ => None,
+            };
+
+            let encrypted = if let Some(shared_secret) = shared_secret {
+                info!(context, "Encrypting symmetrically.");
```

Comment:
Are such logs actually needed or is it a debug log? Probably we should just make it clear that a broadcast message is sent

Reply:
Maybe then log which type of message we're going to send? And also the chat id, grpid. I.e. all the info which is sufficient to understand how the message is encrypted